sdk/include/opentelemetry/sdk/trace/tracer.h

```diff
@@ -23,7 +24,9 @@ class Tracer final : public trace_api::Tracer, public std::enable_shared_from_th
    * nullptr.
    */
   explicit Tracer(std::shared_ptr<SpanProcessor> processor,
-                  std::shared_ptr<Sampler> sampler = std::make_shared<AlwaysOnSampler>()) noexcept;
+                  std::shared_ptr<Sampler> sampler = std::make_shared<AlwaysOnSampler>(),
+                  const opentelemetry::sdk::resource::Resource &resource =
+                      opentelemetry::sdk::resource::Resource::Create({})) noexcept;
```

Comment:
I think this one may leak a nullptr exception from the default value.

I.e. the default argument won't be retained.   AFAIK it should be ok to *not* give this argument a default value and require the TracerProvider to pass one into the tracer.

Reply:
Right, default argument for reference can leak. Have removed default now.